store/sql_post_store.go

```diff
@@ -819,6 +819,16 @@ func (s SqlPostStore) Search(teamId string, userId string, params *model.SearchP
 	go func() {
 		result := StoreResult{}
 
+		if !*utils.Cfg.ServiceSettings.EnablePostSearch {
+			list := &model.PostList{}
+			list.MakeNonNil()
+			result.Data = list
+			result.Err = model.NewLocAppError("SqlPostStore.Search", "Searching has been disabled on this server. Please contact your System Administrator.", nil, fmt.Sprintf("teamId=%v userId=%v params=%v", teamId, userId, params.ToJson()))
```

Comment:
Yes, but this is only for the shift branch.  I'll submit a proper fix to master.